src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -738,6 +739,22 @@ WhereClauseList(FromExpr *fromExpr)
 
 
 /*
+ * OuterJoinClauseList walks over the FROM expression in the query tree, and builds
+ * a list of outer join clauses from the expression tree.
+ */
+List *
+OuterJoinClauseList(FromExpr *fromExpr)
```

Comment:
This is a copy/paste from ```WhereClauseList```  and ```ExtractOuterJoinQualifiersWalker``` is almost identical to ```ExtractFromExpressionWalker```. Perhaps we could merge two such that ```WhereClauseList``` returns join expressions for outer joins too.  However, clauses that are used for filtering inside ON() clause should not be reported as it changes post-join filtering semantics.